lib/pf/constants/role.pm

```diff
@@ -0,0 +1,78 @@
+package pf::constants::role;
+
+=head1 NAME
+
+pf::constants::role - constants for roles
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::constants::role
+
+=cut
+
+use strict;
+use warnings;
+use Readonly;
+
+use Exporter qw(import);
+
+our @EXPORT_OK = qw(
+    @ROLES
+    $REGISTRATION_ROLE
+    $ISOLATION_ROLE
+    $MAC_DETECTION_ROLE
+    $INLINE_ROLE
+    $VOICE_ROLE
+);
+
+=head2 ROLES
+
+Required roles for every switch. Those are reserved words for any additional custom role.
+
+=cut
+
+Readonly::Scalar our $VOICE_ROLE => 'voice';
+Readonly::Scalar our $REGISTRATION_ROLE => 'registration';
+Readonly::Scalar our $ISOLATION_ROLE => 'isolation';
+Readonly::Scalar our $MAC_DETECTION_ROLE => 'macDetection';
+Readonly::Scalar our $INLINE_ROLE => 'inline';
+
+Readonly::Array our @ROLES => (
```

Comment:
Seems like this prevents creating a role that is a reserved name (from my understanding)

Since the voice role will now exist, it doesn't need to be validated there since it will be a duplicate of voice which is already in the DB

One thing we need to make sure though is that the voice role *cannot* be deleted from the admin. Otherwise calls to node_modify($ziBigMac, category => $VOICE_ROLE) will fail and they are there in role.pm